src/cockpit/channels/filesystem.py

```diff
@@ -115,10 +115,17 @@ def do_open(self, options):
 class FsReplaceChannel(Channel):
     payload = 'fsreplace1'
 
+    _umask = None
     _path = None
     _tag = None
     _tempfile = None
 
+    def get_umask(self):
+        if self._umask is None:
+            self._umask = os.umask(0o077)
```

Comment:
Hard no.

If you want to do this, then it happens exactly once, and it happens as a call from the start of `main()` in `bridge.py`, please.

We don't have threads (that we know about, as of today) but burying thread-unsafe code like this to be run semi-regularly deep inside our code is a no-go for me.